Cognite.Simulator.Utils/FileStorageClient.cs

```diff
@@ -29,38 +33,65 @@ public FileStorageClient(HttpClient client, ILogger<FileStorageClient> logger)
         /// Downloads the file from the provided <paramref name="uri"/> and saves
         /// it in the provided <paramref name="filePath"/>
         /// </summary>
-        /// <param name="uri"></param>
-        /// <param name="filePath"></param>
+        /// <param name="uri">URI to download from</param>
+        /// <param name="filePath">Path to save the file</param>
+        /// <param name="failLargeFiles">Set to true to throw an exception if the file is larger than <see cref="LargeFileSize"/></param>
         /// <returns><c>true</c> if success, else <c>false</c></returns>
-        public async Task<bool> DownloadFileAsync(Uri uri, string filePath)
+        public async Task<bool> DownloadFileAsync(Uri uri, string filePath, bool failLargeFiles = false)
         {
             try
             {
-                var response = await _client.GetAsync(uri)
+                var response = await _client.GetAsync(uri, HttpCompletionOption.ResponseHeadersRead)
                     .ConfigureAwait(false);
                 if (!response.IsSuccessStatusCode)
                 {
                     _logger.LogWarning("Failed to download the file: {Message}", response.ReasonPhrase);
                     return false;
                 }
 
+                if (response.Content.Headers.ContentLength > MaxFileDownloadSize)
```

Comment:
Could it be the case that `ContentLength`  is null in the headers? should we add a null check here?

Reply:
it's a nullable long. so yeah, we should add a check